src/Illuminate/Foundation/Auth/ResetsPasswords.php

```diff
@@ -137,7 +137,7 @@ public function showResetForm(Request $request, $token = null)
             return $this->getEmail();
         }
 
-        $email = $request->input('email');
+        $email = urldecode($request->input('email'));
```

Comment:
Aren't the input values stored on the request already decoded? That's how I interpret the second note on the [manual page for $_GET](http://php.net/manual/en/reserved.variables.get.php)
> Note:
> The GET variables are passed through urldecode().